README
NAME

    DateTime::TimeZone::Local::Win32 - Determine the local system's time
    zone on Windows

VERSION

    version 1.80

SYNOPSIS

      my $tz = DateTime::TimeZone->new( name => 'local' );
    
      my $tz = DateTime::TimeZone::Local->TimeZone();

DESCRIPTION

    This module provides methods for determining the local time zone on a
    Windows platform.

NAME

    DateTime::TimeZone::Local::Win32 - Determine the local system's time
    zone on Windows

HOW THE TIME ZONE IS DETERMINED

    This class tries the following methods of determining the local time
    zone:

      * $ENV{TZ}

      It checks $ENV{TZ} for a valid time zone name.

      * Windows Registry

      When using the registry, we look for the Windows time zone and use a
      mapping to translate this to an Olson time zone name.

	* Windows Vista and 2008

	We look in "SYSTEM/CurrentControlSet/Control/TimeZoneInformation/"
	for a node named "/TimeZoneKeyName". If this exists, we use this
	key to look up the Olson time zone name in our mapping.

	* Windows NT, Windows 2000, Windows XP, Windows 2003 Server

	We look in "SOFTWARE/Microsoft/Windows NT/CurrentVersion/Time
	Zones/" and loop through all of its sub keys.

	For each sub key, we compare the value of the key with "/Std"
	appended to the end to the value of
	"SYSTEM/CurrentControlSet/Control/TimeZoneInformation/StandardName"
	. This gives us the English name of the Windows time zone, which we
	use to look up the Olson time zone name.

	* Windows 95, Windows 98, Windows Millenium Edition

	The algorithm is the same as for NT, but we loop through the sub
	keys of "SOFTWARE/Microsoft/Windows/CurrentVersion/Time Zones/"

AUTHORS

      * David Pinkowitz <dapink@cpan.org>

      * Dave Rolsky <autarch@urth.org>

COPYRIGHT AND LICENSE

    Copyright (C) 2007-2014 Dave Rolsky <autarch@urth.org> Copyright (C)
    2014 by David Pinkowitz <dapink@cpan.org>

    This is free software; you can redistribute it and/or modify it under
    the same terms as the Perl 5 programming language system itself.



Changes
1.80    2014-11-16

    - This is the initial release creating a separate distribution for
      DateTime::TimeZone::Local::Win32. As a result, additional changes specific
      to the Windows platform should be made more often.

---------------------------------------------------------------------------------
Earlier changes are a subset of the Win32-related changes from DateTime-TimeZone.
---------------------------------------------------------------------------------

1.78    2014-11-05

    - Fix warnings from Perl 5.8.8 and before. Apparently it doesn't like when
      $SIG{__DIE__} is set to undef. Patch by Peter Rabbitson. GitHub PR #88.


1.77    2014-11-04

    - Updated the mapping of Windows -> IANA time zones. Patch by David
      Pinkowitz. RT #10025.

    - Replaced Class::Load with Module::Runtime, and replaced eval with Try::Tiny.


1.55    2012-11-30

    - Updated mapping from Win32 to Olson time zone names. Patch by Andrey
      Makhnutin. RT #81456.


1.36    2011-08-29

    - This release is based on version 2011i of the Olson database. This release
      includes a new zone for South Sudan (Africa/Juba) and changes for Samoa,
      Kaliningrad (Russia), Alaska, Hawaii, Newfoundland/Labrador (Canada), and
      Resolute Bay.

    - Require ExtUtils::MakeMaker 6.58+. This fixes an installation problem on
      Windows.


1.31    2011-03-18

    - Updates for Win32 time zones. Patch by David Pinkowitz.


1.19    2010-05-10

    - This release is based on version 2010j of the Olson database. This release
      creates a new zone for Mexico (America/Bahia_Banderas).

    - Dateline time zone on Win32 now always maps to a fixed -12:00 zone. Patch by
      David Pinkowitz.


1.17    2010-04-10

    - Updated Win32 to Olson name translation mapping. Patch by David
      Pinkowitz. RT #56445.


1.07    2009-12-24

    - Fixed for local time zone determination on Win32. Our tests broke after the
      December 2009 Cumulative Time Zone Update from Microsoft. Patch by David
      Pinkowitz. RT #52978.


0.98    2009-09-11

    - Fixes for Win32 time zones. Added handling for new Windows time zones,
      thanks to Jim Brunette. This should fix test failures on some Win32 systems.


0.85    2009-03-16

    - This release is based on version 2009c of the Olson database. The
      only changes in this release are for Cuba.

    - Fixes for Win32 with Microsoft's December time zone update. This
      updated added a zone for Mauritius that wasn't accounted for in the
      DateTime::TimeZone::Local::Win32 module. Reported by David
      Pinkowitz. RT #43535.

    - Added a hack to fix some brokenness with Module::Build::Compat that
      causes the compatibility Makefile.PL to fail on really old
      ExtUtils::MakeMaker versions (6.17 and older). RT #43605.


0.8301  2008-11-07

    - Test fixes only. The 19local-win32 tried to skip tests when they're
      run without write access to the registry, but this was done in a
      very broken way.


0.83    2008-10-27

    - This release is based on version 2008i of the Olson database. The
      major changes in this release are for Argentina.

    - Updated the docs on how the local time zone is determined for
      various platforms.


0.7904  2008-08-30

    - Added three more Windows time zone names to the Windows -> Olson
      mapping. This should fix test failures on up-to-date Windows
      boxes. Patch by David Pinkowitz. RT #38735.


0.79    2008-07-29

    - This release is based on version 2008e of the Olson database. The
      major changes in this release are for Mauritius and Central Europe
      (historical changes only).

    - Fixes for the local TZ tests on Win32. Thanks to David Pinkowitz for
      pointing out my mistake.


0.7701  2008-06-03

    - Many fixes and improvements to the tests for local time zone
      handling on Windows. Based on a patch from David Pinkowitz.

    * There are no code changes in this release besides these test
      changes.


0.77     2008-05-27

    - This release is based on version 2008c of the Olson database. The
      major changes in this release are for Morocco, Mongolia, and
      Pakistan.

    - More Windows changes from David Pinkowitz. Determining the local
      time zone on a Win32 system should now work regardless of the
      system's locale. Previously it only worked for English language
      systems. RT #36201.


0.76     2008-05-18

    * Dropped support for Perl 5.005.

    - Made the Win32 local time zone bits work on Vista and Server
      2008. Patch by David Pinkowitz. RT #35733.


0.75     2008-04-25

    - A stray STORABLE_attach method in DT::TZ::Floating caused freezing
      and thawing a floating timezone with Storable to fail
      badly. Reported by Kostas Chatzikokolakis.

    - Changed the local time zone detection on Win32 to only require read
      access to the registry. Patch by David Pinkowitz. RT #35273.


0.6904   2007-11-09

    - Updated the Windows -> Olson conversion table, though it's still
      English-only. Added some more Win32 tests to hopefully help diagnose
      a test failure report from a Win32 tester.


0.6903   2007-11-08

    - The Win32::TieRegistry prereq was requiring version 1.0, which
      doesn't exist. I meant to require just any version. Reported by
      CJM. RT #30580.

    - No code changes in this release.


0.6902   2007-11-05

    - Fixed many sloppy uses of eval blocks which would leave $@ set after
      calling various DT::TZ methods/functions. This could cause unrelated
      error checking to find $@ set to something and log an error or
      die. Reported by Kat.


0.6901   2007-11-02

    - Add Win32::TieRegistry to our prereqs if we're on Win32. Reported by
      Ruud H.G. van Tol.


0.61     2007-02-18

    - Lots of internal changes in the code used to determine a system's
      local time zone.

    - As a result, we now have a much better method for doing this on

Makefile.PL
use lib 'inc';
use Devel::AssertOS qw[MSWin32];

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.024.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker 6.58;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Determine the local system's time zone on Windows",
  "AUTHOR" => "David Pinkowitz <dapink\@cpan.org>, Dave Rolsky <autarch\@urth.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => "6.58"
  },
  "DISTNAME" => "DateTime-TimeZone-Local-Win32",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "DateTime::TimeZone::Local::Win32",
  "PREREQ_PM" => {
    "DateTime::TimeZone::Local" => 0,
    "Try::Tiny" => 0,
    "Win32::TieRegistry" => 0,
    "parent" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "DateTime" => 0,
    "File::Basename" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Sys::Hostname" => 0,
    "Test::More" => "0.88",
    "lib" => 0
  },
  "VERSION" => "1.80",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "DateTime" => 0,
  "DateTime::TimeZone::Local" => 0,
  "ExtUtils::MakeMaker" => "6.58",
  "File::Basename" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Sys::Hostname" => 0,
  "Test::More" => "0.88",
  "Try::Tiny" => 0,
  "Win32::TieRegistry" => 0,
  "lib" => 0,
  "parent" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);





lib/DateTime/TimeZone/Local/Win32.pm
package DateTime::TimeZone::Local::Win32;
$DateTime::TimeZone::Local::Win32::VERSION = '1.80';
use 5.006;

use strict;
use warnings;

use Try::Tiny;
use Win32::TieRegistry ( 'KEY_READ', Delimiter => q{/} );

use parent 'DateTime::TimeZone::Local';

sub Methods { return qw( FromEnv FromRegistry ) }

sub EnvVars { return 'TZ' }

{
    # This list comes (mostly) in the zipball for the Chronos project
    # - a Smalltalk datetime library. Thanks, Chronos!
    my %WinToOlson = (
        'Afghanistan'                     => 'Asia/Kabul',
        'Afghanistan Standard Time'       => 'Asia/Kabul',
        'Alaskan'                         => 'America/Anchorage',
        'Alaskan Standard Time'           => 'America/Anchorage',
        'Arab'                            => 'Asia/Riyadh',
        'Arab Standard Time'              => 'Asia/Riyadh',
        'Arabian'                         => 'Asia/Muscat',
        'Arabian Standard Time'           => 'Asia/Muscat',
        'Arabic Standard Time'            => 'Asia/Baghdad',
        'Argentina Standard Time'         => 'America/Argentina/Buenos_Aires',
        'Armenian Standard Time'          => 'Asia/Yerevan',
        'Atlantic'                        => 'America/Halifax',
        'Atlantic Standard Time'          => 'America/Halifax',
        'AUS Central'                     => 'Australia/Darwin',
        'AUS Central Standard Time'       => 'Australia/Darwin',
        'AUS Eastern'                     => 'Australia/Sydney',
        'AUS Eastern Standard Time'       => 'Australia/Sydney',
        'Azerbaijan Standard Time'        => 'Asia/Baku',
        'Azores'                          => 'Atlantic/Azores',
        'Azores Standard Time'            => 'Atlantic/Azores',
        'Bahia Standard Time'             => 'America/Bahia',
        'Bangkok'                         => 'Asia/Bangkok',
        'Bangkok Standard Time'           => 'Asia/Bangkok',
        'Bangladesh Standard Time'        => 'Asia/Dhaka',
        'Beijing'                         => 'Asia/Shanghai',
        'Belarus Standard Time'           => 'Europe/Minsk',
        'Canada Central'                  => 'America/Regina',
        'Canada Central Standard Time'    => 'America/Regina',
        'Cape Verde Standard Time'        => 'Atlantic/Cape_Verde',
        'Caucasus'                        => 'Asia/Yerevan',
        'Caucasus Standard Time'          => 'Asia/Yerevan',
        'Cen. Australia'                  => 'Australia/Adelaide',
        'Cen. Australia Standard Time'    => 'Australia/Adelaide',
        'Central'                         => 'America/Chicago',
        'Central America Standard Time'   => 'America/Regina',
        'Central Asia'                    => 'Asia/Almaty',
        'Central Asia Standard Time'      => 'Asia/Almaty',
        'Central Brazilian Standard Time' => 'America/Cuiaba',
        'Central Europe'                  => 'Europe/Prague',
        'Central Europe Standard Time'    => 'Europe/Prague',
        'Central European'                => 'Europe/Belgrade',
        'Central European Standard Time'  => 'Europe/Belgrade',
        'Central Pacific'                 => 'Pacific/Guadalcanal',
        'Central Pacific Standard Time'   => 'Pacific/Guadalcanal',
        'Central Standard Time'           => 'America/Chicago',
        'Central Standard Time (Mexico)'  => 'America/Mexico_City',
        'China'                           => 'Asia/Shanghai',
        'China Standard Time'             => 'Asia/Shanghai',
        'Dateline'                        => '-1200',
        'Dateline Standard Time'          => '-1200',
        'E. Africa'                       => 'Africa/Nairobi',
        'E. Africa Standard Time'         => 'Africa/Nairobi',
        'E. Australia'                    => 'Australia/Brisbane',
        'E. Australia Standard Time'      => 'Australia/Brisbane',
        'E. Europe'                       => 'Europe/Helsinki',
        'E. Europe Standard Time'         => 'Europe/Helsinki',
        'E. South America'                => 'America/Sao_Paulo',
        'E. South America Standard Time'  => 'America/Sao_Paulo',
        'Eastern'                         => 'America/New_York',
        'Eastern Standard Time'           => 'America/New_York',
        'Egypt'                           => 'Africa/Cairo',
        'Egypt Standard Time'             => 'Africa/Cairo',
        'Ekaterinburg'                    => 'Asia/Yekaterinburg',
        'Ekaterinburg Standard Time'      => 'Asia/Yekaterinburg',
        'Fiji'                            => 'Pacific/Fiji',
        'Fiji Standard Time'              => 'Pacific/Fiji',
        'FLE'                             => 'Europe/Helsinki',
        'FLE Standard Time'               => 'Europe/Helsinki',
        'Georgian Standard Time'          => 'Asia/Tbilisi',
        'GFT'                             => 'Europe/Athens',
        'GFT Standard Time'               => 'Europe/Athens',
        'GMT'                             => 'Europe/London',
        'GMT Standard Time'               => 'Europe/London',
        'Greenland Standard Time'         => 'America/Godthab',
        'Greenwich'                       => 'GMT',
        'Greenwich Standard Time'         => 'GMT',
        'GTB'                             => 'Europe/Athens',
        'GTB Standard Time'               => 'Europe/Athens',
        'Hawaiian'                        => 'Pacific/Honolulu',
        'Hawaiian Standard Time'          => 'Pacific/Honolulu',
        'India'                           => 'Asia/Calcutta',
        'India Standard Time'             => 'Asia/Calcutta',
        'Iran'                            => 'Asia/Tehran',
        'Iran Standard Time'              => 'Asia/Tehran',
        'Israel'                          => 'Asia/Jerusalem',
        'Israel Standard Time'            => 'Asia/Jerusalem',
        'Jordan Standard Time'            => 'Asia/Amman',
        'Kaliningrad Standard Time'       => 'Europe/Kaliningrad',
        'Kamchatka Standard Time'         => 'Asia/Kamchatka',
        'Korea'                           => 'Asia/Seoul',
        'Korea Standard Time'             => 'Asia/Seoul',
        'Libya Standard Time'             => 'Africa/Tripoli',
        'Line Islands Standard Time'      => 'Pacific/Kiritimati',
        'Magadan Standard Time'           => 'Asia/Magadan',
        'Mauritius Standard Time'         => 'Indian/Mauritius',
        'Mexico'                          => 'America/Mexico_City',
        'Mexico Standard Time'            => 'America/Mexico_City',
        'Mexico Standard Time 2'          => 'America/Chihuahua',
        'Mid-Atlantic'                    => 'Atlantic/South_Georgia',
        'Mid-Atlantic Standard Time'      => 'Atlantic/South_Georgia',
        'Middle East Standard Time'       => 'Asia/Beirut',
        'Montevideo Standard Time'        => 'America/Montevideo',
        'Morocco Standard Time'           => 'Africa/Casablanca',
        'Mountain'                        => 'America/Denver',
        'Mountain Standard Time'          => 'America/Denver',
        'Mountain Standard Time (Mexico)' => 'America/Chihuahua',
        'Myanmar Standard Time'           => 'Asia/Rangoon',
        'N. Central Asia Standard Time'   => 'Asia/Novosibirsk',
        'Namibia Standard Time'           => 'Africa/Windhoek',
        'Nepal Standard Time'             => 'Asia/Katmandu',
        'New Zealand'                     => 'Pacific/Auckland',
        'New Zealand Standard Time'       => 'Pacific/Auckland',
        'Newfoundland'                    => 'America/St_Johns',
        'Newfoundland Standard Time'      => 'America/St_Johns',
        'North Asia East Standard Time'   => 'Asia/Irkutsk',
        'North Asia Standard Time'        => 'Asia/Krasnoyarsk',
        'Pacific'                         => 'America/Los_Angeles',
        'Pacific SA'                      => 'America/Santiago',
        'Pacific SA Standard Time'        => 'America/Santiago',
        'Pacific Standard Time'           => 'America/Los_Angeles',
        'Pacific Standard Time (Mexico)'  => 'America/Tijuana',
        'Pakistan Standard Time'          => 'Asia/Karachi',
        'Paraguay Standard Time'          => 'America/Asuncion',
        'Prague Bratislava'               => 'Europe/Prague',
        'Romance'                         => 'Europe/Paris',
        'Romance Standard Time'           => 'Europe/Paris',
        'Russia Time Zone 10'             => 'Asia/Srednekolymsk',
        'Russia Time Zone 11'             => 'Asia/Anadyr',
        'Russia Time Zone 3'              => 'Europe/Samara',
        'Russian'                         => 'Europe/Moscow',
        'Russian Standard Time'           => 'Europe/Moscow',
        'SA Eastern'                      => 'America/Cayenne',
        'SA Eastern Standard Time'        => 'America/Cayenne',
        'SA Pacific'                      => 'America/Bogota',
        'SA Pacific Standard Time'        => 'America/Bogota',
        'SA Western'                      => 'America/Guyana',
        'SA Western Standard Time'        => 'America/Guyana',
        'Samoa'                           => 'Pacific/Apia',
        'Samoa Standard Time'             => 'Pacific/Apia',
        'Saudi Arabia'                    => 'Asia/Riyadh',
        'Saudi Arabia Standard Time'      => 'Asia/Riyadh',
        'SE Asia'                         => 'Asia/Bangkok',
        'SE Asia Standard Time'           => 'Asia/Bangkok',
        'Singapore'                       => 'Asia/Singapore',
        'Singapore Standard Time'         => 'Asia/Singapore',
        'South Africa'                    => 'Africa/Harare',
        'South Africa Standard Time'      => 'Africa/Harare',
        'Sri Lanka'                       => 'Asia/Colombo',
        'Sri Lanka Standard Time'         => 'Asia/Colombo',
        'Syria Standard Time'             => 'Asia/Damascus',
        'Sydney Standard Time'            => 'Australia/Sydney',
        'Taipei'                          => 'Asia/Taipei',
        'Taipei Standard Time'            => 'Asia/Taipei',
        'Tasmania'                        => 'Australia/Hobart',
        'Tasmania Standard Time'          => 'Australia/Hobart',
        'Tokyo'                           => 'Asia/Tokyo',
        'Tokyo Standard Time'             => 'Asia/Tokyo',
        'Tonga Standard Time'             => 'Pacific/Tongatapu',
        'Turkey Standard Time'            => 'Europe/Istanbul',
        'Ulaanbaatar Standard Time'       => 'Asia/Ulaanbaatar',
        'US Eastern'                      => 'America/Indianapolis',
        'US Eastern Standard Time'        => 'America/Indianapolis',
        'US Mountain'                     => 'America/Phoenix',
        'US Mountain Standard Time'       => 'America/Phoenix',
        'UTC'                             => 'UTC',
        'UTC+12'                          => '+1200',
        'UTC-02'                          => '-0200',
        'UTC-11'                          => '-1100',
        'Venezuela Standard Time'         => 'America/Caracas',
        'Vladivostok'                     => 'Asia/Vladivostok',
        'Vladivostok Standard Time'       => 'Asia/Vladivostok',
        'W. Australia'                    => 'Australia/Perth',
        'W. Australia Standard Time'      => 'Australia/Perth',
        'W. Central Africa Standard Time' => 'Africa/Luanda',
        'W. Europe'                       => 'Europe/Berlin',
        'W. Europe Standard Time'         => 'Europe/Berlin',
        'Warsaw'                          => 'Europe/Warsaw',
        'West Asia'                       => 'Asia/Karachi',
        'West Asia Standard Time'         => 'Asia/Karachi',
        'West Pacific'                    => 'Pacific/Guam',
        'West Pacific Standard Time'      => 'Pacific/Guam',
        'Western Brazilian Standard Time' => 'America/Rio_Branco',
        'Yakutsk'                         => 'Asia/Yakutsk',
        'Yakutsk Standard Time'           => 'Asia/Yakutsk',
    );

    sub FromRegistry {
        my $class = shift;

        my $win_name = $class->_FindWindowsTZName();

        # On Windows 2008 Server, there is additional junk after a
        # null character.
        $win_name =~ s/\0.*$//
            if defined $win_name;

        return unless defined $win_name;

        my $olson = $WinToOlson{$win_name};

        return unless $olson;

        return unless $class->_IsValidName($olson);

        return try {
            local $SIG{__DIE__};
            DateTime::TimeZone->new( name => $olson );
        };
    }
}

sub _FindWindowsTZName {
    my $class = shift;

    my $LMachine = $Registry->Open( 'LMachine/', { Access => KEY_READ } );

    my $TimeZoneInfo = $LMachine->{
        'SYSTEM/CurrentControlSet/Control/TimeZoneInformation/'};

    # Windows Vista, Windows 2008 Server
    if ( defined $TimeZoneInfo->{'/TimeZoneKeyName'}
        && $TimeZoneInfo->{'/TimeZoneKeyName'} ne '' ) {
        return $TimeZoneInfo->{'/TimeZoneKeyName'};
    }
    else {
        my $AllTimeZones = $LMachine->{
            'SOFTWARE/Microsoft/Windows NT/CurrentVersion/Time Zones/'}

            # Windows NT, Windows 2000, Windows XP, Windows 2003 Server
            ? $LMachine->{
            'SOFTWARE/Microsoft/Windows NT/CurrentVersion/Time Zones/'}

            # Windows 95, Windows 98, Windows Millenium Edition
            : $LMachine->{
            'SOFTWARE/Microsoft/Windows/CurrentVersion/Time Zones/'};

        foreach my $zone ( $AllTimeZones->SubKeyNames() ) {
            if ( $AllTimeZones->{ $zone . '/Std' } eq
                $TimeZoneInfo->{'/StandardName'} ) {
                return $zone;
            }
        }
    }

    return;
}

1;

# ABSTRACT: Determine the local system's time zone on Windows

__END__

=pod

=encoding UTF-8

=head1 NAME

DateTime::TimeZone::Local::Win32 - Determine the local system's time zone on Windows

=head1 VERSION

version 1.80

=head1 SYNOPSIS

  my $tz = DateTime::TimeZone->new( name => 'local' );

  my $tz = DateTime::TimeZone::Local->TimeZone();

=head1 DESCRIPTION

This module provides methods for determining the local time zone on a
Windows platform.

=head1 NAME

DateTime::TimeZone::Local::Win32 - Determine the local system's time zone on Windows

=head1 HOW THE TIME ZONE IS DETERMINED

This class tries the following methods of determining the local time
zone:

=over 4

=item * $ENV{TZ}

It checks C<< $ENV{TZ} >> for a valid time zone name.

=item * Windows Registry

When using the registry, we look for the Windows time zone and use a
mapping to translate this to an Olson time zone name.

=over 8

=item * Windows Vista and 2008

We look in "SYSTEM/CurrentControlSet/Control/TimeZoneInformation/" for
a node named "/TimeZoneKeyName". If this exists, we use this key to
look up the Olson time zone name in our mapping.

=item * Windows NT, Windows 2000, Windows XP, Windows 2003 Server

We look in "SOFTWARE/Microsoft/Windows NT/CurrentVersion/Time Zones/"
and loop through all of its sub keys.

For each sub key, we compare the value of the key with "/Std" appended
to the end to the value of
"SYSTEM/CurrentControlSet/Control/TimeZoneInformation/StandardName". This
gives us the I<English> name of the Windows time zone, which we use to
look up the Olson time zone name.

=item * Windows 95, Windows 98, Windows Millenium Edition

The algorithm is the same as for NT, but we loop through the sub keys
of "SOFTWARE/Microsoft/Windows/CurrentVersion/Time Zones/"

=back

=back

=head1 AUTHORS

=over 4

=item *

David Pinkowitz <dapink@cpan.org>

=item *

Dave Rolsky <autarch@urth.org>

=back

=head1 COPYRIGHT AND LICENSE

Copyright (C) 2007-2014 Dave Rolsky <autarch@urth.org>
Copyright (C) 2014 by David Pinkowitz <dapink@cpan.org>

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut
